### PR TITLE
Refactor legacy k truss

### LINKED ISSUE #3753: Implement K-Truss C API to connect to legacy implementation
Define and implement K-Truss C API so that it can be used to call the legacy implementation.  This will enable the python code to be updated to use the new PLC approach.  The legacy python/cython code has some ugly code that can easily be replaced using PLC.

### LINKED ISSUE #3754: Modify K-Truss python to call C API connecting to legacy implementation
Depends on #3753

Modify the python/cython code for K-Truss to use the new C API for connecting to the legacy implementation.

There's some ugly code in the legacy implementation that can be simplified and replaced by using this approach.

### PR DESCRIPTION
**Pull request by jnke2016**
This PR refactor legacy k truss by leveraging the `C API`  instead of the `legacy:COO` graph. This also stands as a preliminary work to later provide a new implementation of `k_truss` matching the `C API`.

Closes #3753 
Closes #3754

### CODE REVIEW

File: cpp/tests/c_api/legacy_k_truss_test.c [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  cugraph_error_free(ret_error);
+
+  return test_ret_value;
+}
+
+int test_k_truss()
+{
+  size_t num_edges    = 16;
+  size_t num_vertices = 6;
+  size_t k = 2;
```
Comment (ChuckHastings) [inline cpp/tests/c_api/legacy_k_truss_test.c:175]: k = 2 is not very interesting, just returns the whole graph.  Perhaps try k = 3?

Comment (jnke2016) [reply cpp/tests/c_api/legacy_k_truss_test.c:175]: I believe it crashes but let me double check. Maybe I should get a more interesting datasets if it is the case?

File: cpp/tests/c_api/legacy_k_truss_test.c [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+            found = (nearlyEqual(h_expected_wgt[i], h_result_wgt[j], 0.001));
+          }
+          else{
+            found = TRUE;
+          }
+    }
+    TEST_ASSERT(test_ret_value, found, "extracted an edge that doesn't match");
+  }
+
+#if 0
```
[the remark this answers is not in the document]
Comment (jnke2016) [reply cpp/tests/c_api/legacy_k_truss_test.c:150]: Right I forgot to delete it

File: cpp/tests/c_api/legacy_k_truss_test.c (hunk 1 above)
Comment (jnke2016) [reply cpp/tests/c_api/legacy_k_truss_test.c:175]: Never mind. it works for 3 and we actually get a subgraph smaller than the initital graph. Let me update this tests accordingly

**Review by ChuckHastings — APPROVED** [review]

**Review by BradReesWork — APPROVED** [review]

File: python/pylibcugraph/pylibcugraph/k_truss_subgraph.pyx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    ...     ,dtype=numpy.float32)   
+    >>> k = 2
+    >>> resource_handle = pylibcugraph.ResourceHandle()
+    >>> graph_props = pylibcugraph.GraphProperties(
+    ...     is_symmetric=True, is_multigraph=False)
+    >>> G = pylibcugraph.SGGraph(
+    ...     resource_handle, graph_props, srcs, dsts, weights,
+    ...     store_transposed=False, renumber=False, do_expensive_check=False)
+    >>> (sources, destinations, edge_weights, subgraph_offsets) =
+    ...     pylibcugraph.k_truss_subgraph(resource_handle, G, k, False)
```
Comment (acostadon) [inline python/pylibcugraph/pylibcugraph/k_truss_subgraph.pyx:102]: I get an error when I execute this line
>>> (sources, destinations, edge_weights, subgraph_offsets) =  pylibcugraph.k_truss_subgraph(resource_handle, G, k, False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'pylibcugraph' has no attribute 'k_truss_subgraph'

**Review by acostadon — APPROVED** [review]

**Pull request merged**